tls/s2n_connection.c

```diff
@@ -81,6 +81,10 @@ static int s2n_connection_init_hmacs(struct s2n_connection *conn)
     return 0;
 }
 
+/* Allocates and initializes memory for a new connection.
+ *
+ * Ensure that all initialization happens in `s2n_connection_wipe` since customers can
+ * reuse a connection. */
```

Comment:
It's not really "all initialization". One-time initialization, like allocating persistent memory, needs to happen here.